.github/workflows/docker_petsc-parmmg.yml

```diff
@@ -1,10 +1,21 @@
 name: 'Build bespoke PETSc Docker container'
 
 on:
-  # Build the Docker container whenever commits are pushed to an open PR
+  # Check that Docker build succeeds when these files get changed in an open PR
   pull_request:
+    paths:
+      - '.github/workflows/docker_petsc-parmmg.yml'
+      - 'docker/Dockerfile.petsc'
 
-  # Build the Docker container every Saturday at 1AM
+  # Build and push the Docker container when these files get changed on the main branch
+  push:
+    branches:
+      - main
+    paths:
+      - '.github/workflows/docker_petsc-parmmg.yml'
+      - 'docker/Dockerfile.petsc'
```

Comment:
So presumably this `paths` syntax restricts so that the Docker image only gets built if the paths have changed?

Reply:
It controls whether the entire workflow will be run or not, so yes :)

Reply:
Ah okay I missed that. Much neater approach then, thanks!